[AND-5244] - JSON support

walter-rover opened the pull request.

walter-rover committed:
[AND-5423] - Allow custom headers to be passed when creating AloomaApi

walter-rover committed:
[AND-5244] - Allow AloomaAPI to send application/json and application/x-www-url-form-encoded data

ExpandedMind commented on src/main/java/com/github/aloomaio/androidsdk/aloomametrics/AnalyticsMessages.java:
```diff
@@ -321,61 +354,56 @@ else if (msg.what == KILL_WORKER) {
                     final String lastId = eventsData[0];
                     final String rawMessage = eventsData[1];
 
-                    final String encodedData = Base64Coder.encodeString(rawMessage);
```
just wondering if we should leave backwards compatibility

walter-rover replied:
Yeah I just moved some of this logic into `HttpService`

walter-rover committed:
[AND-5244] - Fix warnings

walter-rover committed:
[AND-5244] - Remove validateToken function

walter-rover committed:
[AND-5244] - Fix null pointer exception when making request without data

walter-rover committed:
[AND-5244] - If using JSON contentType place all attributes beneaath property object in payload

The IOS, JS and Python SDKs all follow the same request payload schema. The JSON object contains 2 keys, event and properties.
In the properties object any JSON type can be added. This change unifies the Android payload with the other SDKs.

walter-rover committed:
[AND-5244] - Allow multiple instances of AnalyticsMessage

Allow multiple instances of AnalyticsMessages. Each instance is maintains its own worker thread. The worker thread is responsible for adding and removing messages from the sqlite database.
In addition the worker will flush the database periodically sending all messages to an external RemoteService.
